src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -1067,6 +1184,114 @@ public void renderBlockOverlay(DrawBlockHighlightEvent event) {
 							}
 						}
 
+						GlStateManager.depthMask(true);
+						GlStateManager.enableTexture2D();
+						GlStateManager.disableBlend();
+					}
+				} else if ((heldInternal.equals("HOE_OF_GREATEST_TILLING") &&
+					NotEnoughUpdates.INSTANCE.config.itemOverlays.enableHoeOverlay && onPrivateIsland) &&
+					event.target.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK) {
+					BlockPos target = event.target.getBlockPos();
+					IBlockState targetState = Minecraft.getMinecraft().theWorld.getBlockState(target);
+
+					if (targetState.getBlock() == Blocks.dirt || targetState.getBlock() == Blocks.grass) {
+						GlStateManager.enableDepth();
+						GlStateManager.enableBlend();
+						GlStateManager.tryBlendFuncSeparate(770, 771, 1, 0);
+						GlStateManager.disableTexture2D();
+						GlStateManager.depthMask(true);
+
+						LinkedList<BlockPos> candidates = new LinkedList<>();
+						LinkedList<BlockPos> candidatesOld = new LinkedList<>();
+						LinkedList<BlockPos> candidatesNew = new LinkedList<>();
+						candidatesNew.add(target);
+
+						while (candidatesOld.size() <= 100) {
+							if (candidatesNew.isEmpty()) {
+								break;
+							}
+
+							candidates.addAll(candidatesNew);
+							candidatesNew.clear();
+
+							while (!candidates.isEmpty()) {
+
+								BlockPos candidate = candidates.pop();
+								candidatesOld.add(candidate);
+
+								float yaw = Minecraft.getMinecraft().thePlayer.getRotationYawHead();
+								Facing facing = Facing.forDirection(yaw);
+								int xOff = facing == Facing.WEST ? -1 : facing == Facing.EAST ? 1 : 0;
+								int zOff = facing == Facing.NORTH ? -1 : facing == Facing.SOUTH ? 1 : 0;
+
+								BlockPos renderPos = candidate.add(xOff, 0, zOff);
+								IBlockState renderState = Minecraft.getMinecraft().theWorld.getBlockState(renderPos);
+								if (!candidatesOld.contains(renderPos) && !candidates.contains(renderPos) && !candidatesNew.contains(
+									renderPos)) {
+									if (renderState.getBlock() == Blocks.dirt || renderState.getBlock() == Blocks.grass) {
+										candidatesNew.add(renderPos);
+									} else {
+										break;
+									}
+								}
+							}
+
+							for (BlockPos renderPos : candidatesNew) {
+								AxisAlignedBB bbExpanded = Blocks.dirt.getSelectedBoundingBox(
+																									 Minecraft.getMinecraft().theWorld,
```

Comment:
![image](https://user-images.githubusercontent.com/20768569/221385173-11ec8fce-aeb4-4de8-9ba4-11d4b12135bf.png)
im once again complaining about your absolutely shit knowledge about how early returns work, and how methods can help you organize your code. for how long exactly have you been coding now? my little brother is fucking 12 and has been coding for less than half a year and knows what methods are, do better.